pkg/server/commands/list_objects.go

```diff
@@ -355,7 +352,11 @@ func (q *ListObjectsQuery) Execute(
 					return nil, result.Err
 				}
 
-				return nil, serverErrors.HandleError("", result.Err)
+				if !(errors.Is(result.Err, context.Canceled) || errors.Is(result.Err, context.DeadlineExceeded)) {
+					return nil, serverErrors.HandleError("", result.Err)
+				}
+
+				continue
```

Comment:
The assumption here is that if any of the sub-Checks or the ReverseExpand query returns a cancellation or deadline exceeded error, then we assume that either a) the operation was cancelled because this code path has already met the limit and has since returned (e.g. this would be a context.Canceled) or b) the deadline wasn't met.

In these cases we return everything we could find up to that point.